components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/model/EndpointConfig.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.action.management.model;
+
+/**
+ * EndpointConfig.
+ */
+public class EndpointConfig {
+
+    private String uri;
+    private AuthType authentication;
+
+    public EndpointConfig() {
+    }
+
+    public EndpointConfig(EndpointConfigBuilder endpointConfigBuilder) {
+
+        this.uri = endpointConfigBuilder.uri;
+        this.authentication = endpointConfigBuilder.authentication;
+    }
+    public String getUri() {
+
+        return uri;
+    }
+
+    public void setUri(String uri) {
+
+        this.uri = uri;
+    }
+
+    public AuthType getAuthentication() {
+
+        return authentication;
+    }
+
+    public void setAuthentication(AuthType authentication) {
+
+        this.authentication = authentication;
+    }
+
```

Comment:
do we need these setters as the builder is there. If builder is intended to be used better just have it